### ISSUE TITLE
EVA/ClilnVar - Include mode of inheritance

### ISSUE THREAD

**Issue by DSuveges** [opentargets/platform#1140]

**Comment by AsierGonzalez** [comment]
It seems that specifying the mode of inheritance in ClinVar is up to the submitters and [it seems to appear in the variant page](https://www.ncbi.nlm.nih.gov/clinvar/variation/48449/#id_second):

> When a submitter specifies a mode of inheritance, that value is displayed on the variant page on the "Clinical assertions" tab in parentheses in the Conditions column, as in this example 

From info here: https://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/#mode

**Comment by AsierGonzalez** [comment]
We need to check with Kirill if it is possible to extract this information.

**Comment by AsierGonzalez** [comment]
Kirill is working on this as part of his [EBIvariation/eva-opentargets#123](https://github.com/EBIvariation/eva-opentargets/issues/123) ticket.

**Comment by tskir** [comment]
@DSuveges @AsierGonzalez (copying @tcezard)

Yes it's possible to extract this information, but is only appears in a minority of ClinVar records:
![mode-of-inheritance](https://user-images.githubusercontent.com/10669118/88088693-bb7deb00-cb82-11ea-8544-31d907d8df37.png)

Please let me know how would you like to proceed here; especially, how do I specify the inheritance mode in the evidence strings.

**Comment by DSuveges** [comment]
Although this inheritance information is sparse, we still believe it has high value when it is available. I think this also can be added to the `.evidence.variant2disease` object like this:

```json
"variant2disease": {
  "mode_of_inheritance": "Autosomal dominant inheritance"
}
```

**Comment by tskir** [comment]
@DSuveges (copying @tcezard)

Okay, then I'll go on to implement it in the way you described. Two comments:
* The difference between "genetic_association" and "somatic_mutation" was earlier raised in #1137. For the time being, while their structures are different, I can put the fields in `.evidence.variant2disease`/`.evidence`, accordingly, as you described.
* This will also require changes in the JSON schema, again as described in #1137.

**Comment by DSuveges** [comment]
This [json_schema branch](https://github.com/opentargets/json_schema/tree/ds_eva_changes_2) now has the following definition to store mode of inheritance:

```json
                "mode_of_inheritance": {
                  "type": "string",
                  "description": "ClinVar mode of inheritance",
                  "enum": [
                    "autosomal dominant inheritance",
                    "autosomal recessive inheritance",
                    "somatic mutation",
                    "x-linked inheritance",
                    "autosomal unknown",
                    "x-linked recessive inheritance",
                    "sporadic",
                    "mitochondrial inheritance",
                    "x-linked dominant inheritance",
                    "modeofinheritance multiple",
                    "oligogenic inheritance",
                    "other",
                    "y-linked inheritance",
                    "sex-limited autosomal dominant",
                    "autosomal dominant inheritance with maternal imprinting",
                    "genetic anticipation",
                    "co-dominant",
                    "autosomal dominant inheritance with paternal imprinting",
                    "enigma rules, 2015"
                  ]
                }
```

So I have added the complete list seen on Kiril's graph. I am not 100% sure if this list is complete, or should be checked with `enum`. However it seems there are inconsistencies, which would be good to catch (eg. `autosomal unknown` vs `Autosomal unknown`).

**Comment by tskir** [comment]
Thanks @DSuveges! To answer your question:

> I am not 100% sure if this list is complete

It is as of the latest (mid-July) ClinVar data. Of course, in the future they might add new types

> or should be checked with enum

Yes, this is a difficult choice. I guess we can go with an enum for now, and if they keep changing the list of allowed values too frequently, we can think of other ideas.

**Comment by DSuveges** [comment]
Thanks. Let's try to enumerate the terms and review later.

**Comment by AsierGonzalez** [comment]
Using an enum here is overkill in my opinion. As you have said, if the list changes in the future we will need to check it an update it accordingly. The enum would be justified if we needed the mode of inheritance to take a value from a closed list for instance because we are going to use it to score. However, we are far from there because there are at least three other data sources (Genomics England PanelApp, Gene2Phenotype and ClinGen) that contain similar information but with different possible values (table kindly provided by @emcdonagh-OT):

| PanelApp (list) | Gene2Phenotype | ClinGen |
| --- | --- | --- |
| MONOALLELIC,   autosomal or pseudoautosomal, not imprinted | | |
| MONOALLELIC, autosomal or pseudoautosomal, maternally imprinted (paternal allele expressed) | | |
| MONOALLELIC, autosomal or pseudoautosomal, paternally imprinted (maternal allele expressed)| | |
| MONOALLELIC, autosomal or pseudoautosomal, imprinted status unknown | | |
| | | |
| | | |
| | | |
| | | |
| | | |

**Comment by AsierGonzalez** [comment]
Also, I wonder what the right place for this information is. As you already said, for `somatic_mutation` evidence the mode of inheritance (`inheritance_pattern`) is located directly under `evidence` because that is all the evidence strings contain apart from the target and disease information. However, I am not so sure about placing it under `variant2disease` for genetics associations. Is the mode of inheritance an inherent property of the variant in that disease or is it related to the target or the variant? If it's the former, then the suggested location is fine, if it's the latter it should be a property of the target or the variant.

**Comment by tskir** [comment]
@AsierGonzalez 

Having examined your arguments, I now agree that we should drop enum for mode of inheritance. I will also not do any attempts to format the values of that field (to lowercase or otherwise), and will just report them verbatim.

Regarding your second question: as far as I know, mode of inheritance is usually considered to be a property of the disease. For example, people say that mode of inheritance of Hemophilia A is X-linked recessive. However, it is mostly influenced by the properties of the gene: where it is located (autosome, allosome, or mitochondrial genome) and how variants in that gene (especially LOF ones) impact the phenotype associated with the disease, based on their zygosity. I also believe there are some rare cases where different variants in the same gene might have different modes of inheritance for the same disease.

It might be a good idea to summon someone with a better understanding of biology and clinical genetics. As far as I see, given that mode of inheritance depends on all three concepts (variant, disease, gene), using the `variant2disease` section seems to be a sensible approach.

**Comment by tskir** [comment]
@AsierGonzalez @DSuveges 

Also, as I see, currently in the JSON schema the `mode_of_inheritance` block is defined only once, inside `variant2disease`. Meaning, once again, that I will only be able to include it for `genetic_association` evidence strings. Is this intended?

**Comment by tskir** [comment]
This is now also implemented, undergoing our internal review (https://github.com/EBIvariation/eva-opentargets/pull/149), and also I submitted another PR with changes to the OT JSON schema (opentargets/json_schema/pull/97).

**Comment by tskir** [comment]
While testing the change, I stumbled upon the fact that there were a few dozen records which have _multiple_ “mode of inheritance” attributes specified. In fact, they are specified in the Sankey diagram above as “ModeOfInheritance multiple”, I just forgot about them up until the testing phase.

For now, I solved this problem by amending the PR for the JSON schema to accept _arrays_ in the mode_of_inheritance attributes as well, and by modifying our pipeline to always store those attributes in an array. If needed, we can discuss this and this behaviour can be changed.

**Comment by AsierGonzalez** [comment]
> It might be a good idea to summon someone with a better understanding of biology and clinical genetics. As far as I see, given that mode of inheritance depends on all three concepts (variant, disease, gene), using the variant2disease section seems to be a sensible approach.

@iandunham has confirmed that this is how it should be, that is, the mode of inheritance is a property of all those together. So there is no need to relocate it.

> Also, as I see, currently in the JSON schema the mode_of_inheritance block is defined only once, inside variant2disease. Meaning, once again, that I will only be able to include it for genetic_association evidence strings. Is this intended?

This was probably an oversight and we will fix it when we work on improving the alignment between the genetic associations and somatic mutations. If you are constrained by the JSON schema to generate the evidence, that is, if you can only represent fields that are explicitly defined in it, just leave it out of the somatic evidence strings for now. However, if you are flexible and you can add it to `.evidence` in the latter group, that would be useful for us to explore the data.

> For now, I solved this problem by amending the PR for the JSON schema to accept arrays in the mode_of_inheritance attributes as well, and by modifying our pipeline to always store those attributes in an array. If needed, we can discuss this and this behaviour can be changed.

Specifying the mode of inheritance in an array is a solution but given than only 33 out of the 35,009 records have multiple modes of inheritance I would suggest that first we check what they represent and whether we could handle them in an alternative way. The mode of inheritance will be provided by other data sources so we should try to make it as generic as possible. If an array is finally required we may need to include it in a `oneOf` construct but we could do that ourselves.

**Comment by tskir** [comment]
@AsierGonzalez Thank you for the detailed comments. Now to address them:

### Multiple ModeOfInheritance attributes

So that we don't have to decide blindly, here is the distribution of all cases where a ClinVar record has multiple ModeOfInheritance attributes. You may notice that the total is 42 instead of 33—this is because I'm using a newer ClinVar data release than the one in https://github.com/opentargets/platform/issues/1140#issuecomment-662010894.

```
     10 Autosomal dominant inheritance|Sporadic
      7 Autosomal dominant inheritance|Autosomal recessive inheritance
      6 Autosomal dominant inheritance|Somatic mutation
      5 Autosomal dominant inheritance|Autosomal unknown
      4 X-linked dominant inheritance|X-linked inheritance
      3 Autosomal recessive inheritance|autosomal unknown
      2 Sporadic|X-linked inheritance
      2 X-linked inheritance|X-linked recessive inheritance
      1 Autosomal recessive inheritance|Autosomal unknown
      1 Autosomal recessive inheritance|X-linked recessive inheritance
      1 Autosomal unknown|Sporadic
```

It looks to me that those combinations of values look quite sensible and should be represented in the evidence strings. Even though those cases are a minority, we should still try to represent all possible data states. (Case in point: having multiple clinical significance values is also a minority, but we support an array of them anyway.)

I agree that in this case it's better to have a `oneOf` choice between an array and a string, to make it easier for other submitters. However, I think that for consistency, it would be better if our pipeline always used the array, even when there is only one value (again, similarly to how we always use an array for clinical significance levels, even when it's just one).

### Alignment with JSON schema changes

Just to clarify, in our pipeline we are _not_ restricted to only specifying the values currently present in the OT JSON schema. However, I believe it is a good practice to introduce changes to the code & the schema at the same time, to avoid significant divergence between them. It's not difficult for me at all to introduce those changes and to submit the corresponding PRs.

**Comment by tskir** [comment]
This is now reviewed & merged on our side. The current implementation is as described in my previous comment: specifying the modes of inheritance both for `genetic_association` and `somatic_mutation` evidence strings, and always using an array.

I also updated https://github.com/opentargets/json_schema/pull/97 accordingly.

**Comment by AsierGonzalez** [comment]
Thank you for breaking the combinations down @tskir. You say that the combinations of values look sensible but I struggle to understand how the same disease caused by the same mutation can have multiple modes of inheritance. Is this biologically possible or a way to represent that there is uncertainty about it?

I am just trying to be sure that this is not low confidence data that should be ignored. Otherwise you solution looks good.

**Comment by AsierGonzalez** [comment]
Comment by @iandunham:

> Some of these dual ones look like they are just evolving knowledge. i.e. the mode has got cleared up. The X linked ones are weird - could be that the dominant mode was given for a male case and recessive for a female case. Autosomal recessive inheritance|X-linked recessive inheritance is just a mess as it shouldn't be able to be both. This all looks like it requires some detailed investigation of the specific cases 

So I think we should look into some of those more in detail. Could you provide the ids of those ClinVar entries so that we can look into them?

**Comment by tskir** [comment]
@AsierGonzalez @iandunham Sure, here is the complete table, extracted from the latest ClinVar XML as of 2020-09-16:

<details>
  <summary>All cases of multiple ModeOfInheritance attributes</summary>
  
RCV ID|Modes of inheritance
-----|-----
RCV000063117|Autosomal dominant inheritance; Autosomal recessive inheritance
RCV000190094|Autosomal dominant inheritance; Autosomal recessive inheritance
RCV000275624|Autosomal dominant inheritance; Autosomal recessive inheritance
RCV000399339|Autosomal dominant inheritance; Autosomal recessive inheritance
RCV000918466|Autosomal dominant inheritance; Autosomal recessive inheritance
RCV000948503|Autosomal dominant inheritance; Autosomal recessive inheritance
RCV001096156|Autosomal dominant inheritance; Autosomal recessive inheritance
RCV000086943|Autosomal dominant inheritance; Autosomal unknown
RCV000087353|Autosomal dominant inheritance; Autosomal unknown
RCV000087631|Autosomal dominant inheritance; Autosomal unknown
RCV000344502|Autosomal dominant inheritance; Autosomal unknown
RCV000495606|Autosomal dominant inheritance; Autosomal unknown
RCV000109523|Autosomal dominant inheritance; Somatic mutation
RCV000287932|Autosomal dominant inheritance; Somatic mutation
RCV000372784|Autosomal dominant inheritance; Somatic mutation
RCV000507857|Autosomal dominant inheritance; Somatic mutation
RCV000507858|Autosomal dominant inheritance; Somatic mutation
RCV001025520|Autosomal dominant inheritance; Somatic mutation
RCV000141918|Autosomal dominant inheritance; Sporadic
RCV000344359|Autosomal dominant inheritance; Sporadic
RCV000346120|Autosomal dominant inheritance; Sporadic
RCV000919753|Autosomal dominant inheritance; Sporadic
RCV000963036|Autosomal dominant inheritance; Sporadic
RCV000966069|Autosomal dominant inheritance; Sporadic
RCV001139707|Autosomal dominant inheritance; Sporadic
RCV001141206|Autosomal dominant inheritance; Sporadic
RCV001455265|Autosomal dominant inheritance; Sporadic
RCV001741488|Autosomal dominant inheritance; Sporadic
RCV000086819|Autosomal recessive inheritance; autosomal unknown
RCV000086873|Autosomal recessive inheritance; autosomal unknown
RCV000087680|Autosomal recessive inheritance; autosomal unknown
RCV000077525|Autosomal recessive inheritance; Autosomal unknown
RCV001954859|Autosomal recessive inheritance; X-linked recessive inheritance
RCV000350399|Autosomal unknown; Sporadic
RCV000070093|Sporadic; X-linked inheritance
RCV000146966|Sporadic; X-linked inheritance
RCV000070104|X-linked dominant inheritance; X-linked inheritance
RCV000398754|X-linked dominant inheritance; X-linked inheritance
RCV000401226|X-linked dominant inheritance; X-linked inheritance
RCV001647962|X-linked dominant inheritance; X-linked inheritance
RCV000087558|X-linked inheritance; X-linked recessive inheritance
RCV000529121|X-linked inheritance; X-linked recessive inheritance
</details>

**Comment by AsierGonzalez** [comment]
I have looked into those clinvar accessions and what I have seen does not look very encouraging. Summary is below but you can see all the information in [the spreadsheet](https://docs.google.com/spreadsheets/d/1BxoTFmGJWCVYUJwjjdOnOeV2ZC_RaeUNP2eRDkku78E/edit?usp=sharing):

- Only 11 out of the 42 accessions (highlighted in green) look like plausible variant-disease associations that could have multiple modes of inheritance.
- In 9 accessions (highlighted in orange) there is at least one of the two modes of inheritance that seems realistic based on variant location, the disease or the allele origin.
- 4 accessions have X-linked related modes of inheritance even though the variants are not in the X chromosome
- 14 accessions should not appear in the evidence file because the condition is missing.
- The ClinVar page does not exist for 4 accessions.

There are three things that we could check to find out how widespread this issue is:

- The number of ClinVar accessions where the condition is missing (usually they have the value ["not provided"-MedGen:CN517202](https://www.ncbi.nlm.nih.gov/medgen/CN517202) or ["not specified"-MedGen:CN169374](https://www.ncbi.nlm.nih.gov/medgen/CN169374)).
- The distribution of the modes of inheritance values per allelic origin to see if many somatic variants have non-somatic inheritance values and vice versa.
- The distribution of the modes of inheritance values per chromosome (information available in `Cytogenetic location`) to see if variants in autosomes get X-linked related inheritance values.

However, considering that we are not planning to use the mode of inheritance for scoring but just to display it in the evidence table, we may just want to use it as is.

**Comment by tskir** [comment]
@AsierGonzalez (also copying @tcezard)

Thank you for looking into this! I have two considerations on the matter:
1. Given that there are legitimate cases where multiple modes of inheritance can be present for the same record, I propose that we leave the current way of reporting them in our pipeline unchanged. That is, to continue always reporting an _array_ of values.
2. While at some point we should check all those things you mentioned, it would take a fair bit of time. Given that modes of inheritance will not yet be used for scoring, I propose that we add those investigations to the backlog to be addressed at some point in the future. I've created a ticket on our side: https://github.com/EBIvariation/eva-opentargets/issues/157

**Comment by AsierGonzalez** [comment]
I agree, let's use it as is for now.
